static/js/util/profile_edit.js

```diff
@@ -125,7 +126,79 @@ export function boundDateField(keySet, label) {
     <span className="validation-error-text">{_.get(errors, keySet)}</span>
   </div>;
 }
+export function boundMonthField(keySet, label) {
+  const {updateProfile, errors, profile} = this.props;
+  let onChange = month => {
+    let clone = _.cloneDeep(profile);
+
+    let current_value = _.get(clone, keySet);
+        // format as ISO-8601
+    let date_value;
+    if(!month){
+      date_value = "";
+    }else if (current_value){
+      date_value = moment(current_value).set('month', month.value).format(DATE_FORMAT);
+    } else {
+      date_value = moment().set('day', 1).set('month', month.value).format(DATE_FORMAT);
+    }
+    _.set(clone, keySet, date_value);
+    updateProfile(clone);
+  };
+  let month = _.get(profile, keySet)? moment(_.get(profile, keySet), DATE_FORMAT).month() : "";
+  const month_options = [
+    { value: 0, label: 'January'},
+    { value: 1, label: 'February'},
+    { value: 2, label: 'March'},
+    { value: 3, label: 'April'},
+    { value: 4, label: 'May'},
+    { value: 5, label: 'June'},
+    { value: 6, label: 'July'},
+    { value: 7, label: 'August'},
+    { value: 8, label: 'September'},
+    { value: 9, label: 'October'},
+    { value: 10, label: 'November'},
+    {value: 11, label: 'December'}
+  ];
+
+  return <div>
+    <Select
+      options={month_options}
+      value={month}
+      placeholder={label}
+      onChange={onChange}
+    />
+    <span className="validation-error-text">{_.get(errors, keySet)}</span>
+  </div>;
+}
+
+export function boundYearField(keySet, label) {
+  const {updateProfile, errors, profile} = this.props;
+  let onChange = year => {
+    let clone = _.cloneDeep(profile);
+    let current_value = _.get(clone, keySet);
+    // format as ISO-8601
+    let date_value;
+    if (!year) {
+      date_value = "";
```

Comment:
This won't work with `date_value.isValid()` below. Can you use some value which is invalid, for example `moment("")`?